models/Socrata.js

```diff
@@ -93,122 +90,177 @@ var Socrata = function (koop) {
-                      callback(null, [{status: 'processing'}])
-
-                      // create GeoJSON return object
-                      retGeoJSON = geojson
-                      // detrmine count of table and needed pages
-                      var count, pages, pageUrls = [],
-                        countUrl = host + socrata.socrata_path + urlid + '.json?$select=count(*)'
-
-                      socrata.request(countUrl, function (err, data, response) {
-                        if (err) {
-                          return callback(err)
-                        }
-                        count = parseInt(JSON.parse(data.body)[0].count, 10)
-                        if ((count / limit) % 1 === 0) {
-                          pages = (count / limit - 1)
-                        } else {
-                          pages = Math.floor(count / limit)
-                        }
-                        // page through data
-                        for (var p = 1; p <= pages; p++) {
-                          var page = [
-                            host,
-                            socrata.socrata_path,
-                            urlid,
-                            '.json?$order=:id&$limit=',
-                            limit,
-                            '&$offset=',
-                            (p * limit)
-                          ].join('')
-                          pageUrls.push(page)
-                        }
-
-                        koop.Cache.getInfo(table, function (err, info) {
-                          if (err) {
-                            console.log('Could not get info', err)
-                            info = {}
-                          }
-                          info.status = 'processing'
-                          koop.Cache.updateInfo(table, info, function () {
-                            pageQueue.push(pageUrls, function () {})
-                          })
-                        })
-                      })
-                    }
+                    koop.log.error(page, err)
                   }
                 })
               })
-            } catch (e) {
-              koop.log.error('Unable to parse response %s', url)
-              callback(e, null)
-            }
-          }
+            })
+          })
         })
+      })
+    }
+
+    // attempt to load from cache, if error perform new request
+    koop.Cache.get(type, key, options, function (err, entry) {
+      if (err) {
+        // Return processing immediately before doing other work
+        callback(null, [{status: 'processing'}])
+        // Simultaneously gather count and metadata by adding them to the queue with their variables
+        infoQueue.push({
+            task: socrata.getRowCount,
+            host: host,
+            urlId: urlId
+        }, function (err, rowCount) {
+              if (err) {
+                // set status to failed
+                return
```

Comment:
@dmfenton I dont know enough about all the changes to understand but I'll pull it down in a bit an look and we hang to talk about it.